config-model/src/main/java/com/yahoo/vespa/model/search/SearchNode.java

```diff
@@ -139,22 +137,13 @@ public NodeSpec getNodeSpec() {
     }
 
     /**
-     * Returns the connection spec string that resolves to this search node.
-     *
-     * @return The connection string.
-     */
-    public String getConnectSpec() {
-        return "tcp/" + getHost().getHostName() + ":" + getRpcPort();
-    }
-
-    /**
      * Returns the number of ports needed by this service.
      *
      * @return The number of ports.
      */
     @Override
     public int getPortCount() {
-        return 5;
+        return 3;
```

Comment:
Port number changes might cause issues during version upgrade, where some nodes run with old config and some nodes run with new config.